brushfire-features/src/main/scala/com/stripe/brushfire/features/FeatureEncoder.scala

```diff
@@ -0,0 +1,27 @@
+package com.stripe.brushfire
+package features
+
+// sealed trait FeatureValue
+// object FeatureValue {
+//   case class FString(value: String) extends FeatureValue
+//   case class FDouble(value: Double) extends FeatureValue
+//   sealed abstract class FBoolean(val value: Boolean) extends FeatureValue
+//   case object FTrue extends FBoolean(true)
+//   case object FFalse extends FBoolean(false)
+//   case object FNull extends FeatureValue
+// }
+
+sealed trait FeatureType
+object FeatureType {
+  case object Ordinal extends FeatureType
+  case object Nominal extends FeatureType
+  case object Continuous extends FeatureType
+  case object Sparse extends FeatureType
+}
+
+case class FeatureMapping[A](
+  mapping: Map[String, (FeatureType, A => FeatureValue)]
```

Comment:
Why do we even need `FeatureType`? And if we don't need it, I think this would be much better as `(Map[String,A] => Map[String,FeatureValue])`.